### PR TITLE
adds UpdateContained function to overlay cache

### PR DESCRIPTION
**Pull request by navillasa**
What:
- Adds UpdateContained function to overlay cache and a test for it

Why:
- The existing functions for updating a node dossier (e.g. UpdateNodeInfo) did not allow a way to update a node's `contained` field, which will be needed for containment mode to work.

Thanks for submitting a PR!

## Code Review Checklist (to be filled out by reviewer)
 - [ ] Does the PR describe what changes are being made?
 - [ ] Does the PR describe why the changes are being made?
 - [ ] Does the code follow [our style guide](https://github.com/storj/docs/blob/master/code/Style.md)?
 - [ ] Does the code follow [our testing guide](https://github.com/storj/docs/blob/master/code/Testing.md)?
 - [ ] Could the PR be broken into smaller PRs?
 - [ ] Does the new code have enough tests? (*every* PR should have tests or justification otherwise. Bug-fix PRs especially)
 - [ ] Does the new code have enough documentation that answers "how do I use it?" and "what does it do?"? (both source documentation and [higher level](https://github.com/storj/docs), diagrams?)
 - [ ] Does any documentation need updating?

### CODE REVIEW

File: pkg/overlay/cache.go [hunk 1]
```diff
@@ -273,6 +275,12 @@ func (cache *Cache) UpdateNodeInfo(ctx context.Context, node storj.NodeID, nodeI
 	return cache.db.UpdateNodeInfo(ctx, node, nodeInfo)
 }
 
+// UpdateContained updates node's contained status in node dossier
+func (cache *Cache) UpdateContained(ctx context.Context, nodeID storj.NodeID, isContained bool) (err error) {
```
Comment (VinozzZ) [inline pkg/overlay/cache.go]: Is this a convention to use named `err` in the return value? I don't see it getting used in this function tho

Comment (navillasa) [reply pkg/overlay/cache.go]: yeah it has to be named to get used by `defer mon.Task()(&ctx)(&err)`, which tracks data about what functions get called https://github.com/spacemonkeygo/monkit

File: pkg/overlay/cache_test.go [hunk 2]
```diff
@@ -94,6 +95,46 @@ func testCache(ctx context.Context, t *testing.T, store overlay.DB) {
...
 
+func TestUpdateContained(t *testing.T) {
+	testplanet.Run(t, testplanet.Config{
+		SatelliteCount: 1, StorageNodeCount: 6, UplinkCount: 1,
+	}, func(t *testing.T, ctx *testcontext.Context, planet *testplanet.Planet) {
+
+		cache := planet.Satellites[0].DB.OverlayCache()
+
+		nodeID0 := planet.StorageNodes[0].ID()
+		nodeID1 := planet.StorageNodes[1].ID()
```
Comment (ifraixedes) [inline pkg/overlay/cache_test.go]: Could you move this declaration down, close to its first usage?

Comment (navillasa) [reply pkg/overlay/cache_test.go]: sure!

**Comment by navillasa** [comment]
this functionality is now incorporated into #1972

**Pull request closed without merging**